server/stream.go

```diff
@@ -1694,6 +1700,18 @@ func (mset *stream) updateWithAdvisory(config *StreamConfig, sendAdvisory bool)
 		// a subsequent update to an existing tier will then move from existing past tier to existing new tier
 	}
 
+	// In the event that some of the stream-level limits have changed, update consumers.
+	if ocfg.LimitInactiveThreshold != cfg.LimitInactiveThreshold ||
+		ocfg.LimitMaxAckPending != cfg.LimitMaxAckPending {
+		for _, c := range mset.consumers {
+			c.mu.RLock()
+			ccfg := c.cfg
+			c.mu.RUnlock()
+			enforceConsumerConfigLimitsFromStream(&ccfg, cfg)
```

Comment:
wow, this is a huge red flag here.  We cannot go changing consumer configs behind users backs.  What if they have client code that creates these consumers there's going to be forever fights.  Same concerns as mentioned earlier really.  But updating created consumers out of users control is like at a whole different level of red flag to me